consensus/obcpbft/pbft-core.go

```diff
@@ -439,25 +440,52 @@ func (instance *pbftCore) committed(digest string, v uint64, n uint64) bool {
 // =============================================================================
 
 // handle new consensus requests
+func (instance *pbftCore) requestSync(msgPayload []byte, senderID uint64) error {
+	msg := &Message{&Message_Request{&Request{Payload: msgPayload,
+		ReplicaId: senderID}}}
+	instance.manager.inject(pbftMessageEvent{
```

Comment:
Whatever is returned by `processEvent` is the _next_ event to be processed (regardless of any other events which might be 'in queue').  This is equivalent to directly calling the function, but without the tight coupling it creates.  So, today, for batch to inject an event into PBFT, it would simply return one from `processEvent`.

As you discuss later, we may want to split these into distinct state machines, rather than stacked ones.  If that were the case, then some modification would likely be required, maybe the return would need to include the target state machine, or maybe some other measure, but it doesn't seem insurmountable.